### PR TITLE
Check if table has qualified column

### PR DESCRIPTION
**Pull request by 4ertus2**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):
Check if qualified column present in the table.

### CODE REVIEW

**Comment by 4ertus2** [comment]
#6836

File: dbms/src/Interpreters/IdentifierSemantic.cpp [hunk 1]
```diff
@@ -132,6 +132,15 @@ std::pair<String, String> IdentifierSemantic::extractDatabaseAndTable(const ASTI
     return { "", identifier.name };
 }
 
+std::optional<String> IdentifierSemantic::extractNestedName(const ASTIdentifier & identifier, const String & table_name)
+{
+    if (identifier.name_parts.size() == 3 && table_name == identifier.name_parts[0])
+        return identifier.name_parts[1] + '.' + identifier.name_parts[2];
```
Comment (alexey-milovidov) [inline dbms/src/Interpreters/IdentifierSemantic.cpp]: How it supposed to work when identifier contains reserved characters that require backquotes?

Comment (4ertus2) [reply dbms/src/Interpreters/IdentifierSemantic.cpp]: The idea is ASTIdentifier place name parts as needed for future usage. If not we have troubles with backquotes not only here but all over the code that uses ASTIdentifier. So we need general conversion functions that can set and compare any mix of quoted and not quoted names. There's something special in Nested namespace but according to its name it expected to be used only for nested columns. While we need to wrap all operations with names.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by nikitamikhaylov** [comment]
Conflicts with 19.16

**Pull request merged**